Ginger/GingerCoreNET/Application Models/Learn/APIModels/ApiActionConversionUtils.cs

```diff
@@ -125,53 +133,141 @@ private void ConvertAction(bool parameterizeRequestBody, bool pullValidations, A
         /// <param name="applicationModel"></param>
         /// <param name="act"></param>
         /// <returns></returns>
-        private ActWebAPIModel GetNewAPIModelAction(ApplicationAPIModel applicationModel, Act act)
+        private ActWebAPIModel GetNewAPIModelAction(ApplicationAPIModel applicationModel, Act act, Dictionary<System.Tuple<string, string>, List<string>> optionalValuesPulledFromConvertedAction)
         {
             AutoMapper.MapperConfiguration mapConfigUIElement = new AutoMapper.MapperConfiguration(cfg => { cfg.CreateMap<Act, ActWebAPIModel>(); });
             ActWebAPIModel newActModel = mapConfigUIElement.CreateMapper().Map<Act, ActWebAPIModel>(act);
             newActModel.APImodelGUID = applicationModel.Guid;
             newActModel.Active = true;
-            newActModel.ActAppModelParameters = applicationModel.AppModelParameters;
+            newActModel.Reset();
+            newActModel.APIModelParamsValue = GetAPIActionParams(applicationModel.AppModelParameters, optionalValuesPulledFromConvertedAction);
+            //newActModel.ReturnValues = GetAPIModelActionReturnValues(applicationModel.ReturnValues); //For now we want to keep original validations on action
             return newActModel;
         }
 
+        private ObservableList<EnhancedActInputValue> GetAPIActionParams(ObservableList<AppModelParameter> paramsList, Dictionary<System.Tuple<string, string>, List<string>> optionalValuesPulledFromConvertedAction)
+        {
+            ObservableList<EnhancedActInputValue> enhancedParamsList = new ObservableList<EnhancedActInputValue>();
+            foreach (AppModelParameter apiModelParam in paramsList)
+            {
+                if (apiModelParam.RequiredAsInput == true)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4513408)